### ISSUE TITLE
Make plugin safe for parallel builds

### ISSUE THREAD

**Issue by drub0y** [vandmo/dependency-lock-maven-plugin#75]
If you run the plugin (`v1.0.1`) with a concurrent build you will be warned that the plugin is not marked as thread-safe and you absolutely will get errors from it whether you're running `check` or `lock`.

```plain
[WARNING] *****************************************************************
[WARNING] * Your build is requesting parallel execution, but this         *
[WARNING] * project contains the following plugin(s) that have goals not  *
[WARNING] * marked as thread-safe to support parallel execution.          *
[WARNING] * While this /may/ work fine, please look for plugin updates    *
[WARNING] * and/or request plugins be made thread-safe.                   *
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against Apache Maven.                           *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked as thread-safe in my-project:
[WARNING]   se.vandmo:dependency-lock-maven-plugin:1.0.1
```

The request is to upgrade the plugin to be parallel build aware.

**Comment by vandmo** [comment]
Have you gotten any errors related to parallel builds? Is there anyway I can reproduce it myself?

The mojos seems relatively thread safe except for usage of MessageDigest
I had a look at https://cwiki.apache.org/confluence/display/MAVEN/Parallel+builds+in+Maven+3 and:
* I didn't find any of the known problematic dependencies
* I am assuming that `MavenProject#getArtifacts()` is thread safe
* The check mojo and the lock mojo can **not** be run in parallel since the lock file is not written atomically. It would not make any sense to try to run them in parallel either since that would result in a non deterministic outcome even if the write would be atomic. I am not sure how to express that in the Mojos though.
* QName should be thread safe
* There are no other static variables than MessageDigest and QName

**Comment by vandmo** [comment]
@drub0y could you try out 0.0.eb2d3a63059dc8bdfc5118ebc2f978bb01b4dfdf ?

**Comment by vandmo** [comment]
Fixed in version 1.0.2
